server/schedule/placement/fit.go

```diff
@@ -110,5 +113,148 @@ func compareRuleFit(a, b *RuleFit) int {
 
 // FitRegion tries to fit peers of a region to the rules.
 func FitRegion(stores core.StoreSetInformer, region *core.RegionInfo, rules []*Rule) *RegionFit {
-	return nil
+	peers := prepareFitPeers(stores, region)
+
+	var regionFit RegionFit
+	if len(rules) == 0 {
+		return &regionFit
+	}
+	for _, rule := range rules {
+		rf := fitRule(peers, rule)
+		regionFit.RuleFits = append(regionFit.RuleFits, rf)
+		// Remove selected.
+		peers = filterPeersBy(peers, func(p *fitPeer) bool {
```

Comment:
the filter actually does `select`, maybe as rust  [iterator](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter).

Reply:
The issue is already discussed here: https://github.com/pingcap/pd/pull/1904#pullrequestreview-312404538

Rust's `filter()` is the same as my examples: it returns items that can pass predicate function.